packages/fluxible-plugin-devtools/src/lib/devtools-plugin.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright 2016, Yahoo! Inc.
+ * Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
+ */
+'use strict';
+import debugLib from 'debug';
+const debug = debugLib('Fluxible:DevToolsPlugin');
+
+/**
+ * Creates a new devtools plugin instance
+ * @returns {DevToolsPlugin}
+ */
+export default function devToolsPlugin() {
+    /**
+     * @class DevToolsPlugin
+     */
+    return {
+        /**
+         * @property {String} name Name of the plugin
+         */
+        name: 'DevToolsPlugin',
+        /**
+         * Called to plug the FluxContext
+         * @method plugContext
+         * @param {Object} contextOptions options passed to the createContext method
+         * @param {Object} contextOptions.debug if true, will enable debug mode and allow fluxible to collect metadata for debugging
+         * @returns {Object}
+         */
+        plugContext: function plugContext(contextOptions, context) {
+            let enableDebug = typeof contextOptions.debug !== 'undefined' ? contextOptions.debug : false;
+            let actionHistory = [];
+            /**
+             * extends the input object with a `devtools` namespace which has
+             * debugging methods available to it.
+             * @method provideDevTools
+             * @param {Object} obj arbitrary input object
+             * @return {void}
+             */
+            function provideDevTools (obj) {
```

Comment:
Needs to be inside. This function is a closure with access to the `actionHistory` var. Otherwise, we would just repeat the contents of the function multiple times.